encoding/json/inbound.go

```diff
@@ -41,9 +41,12 @@ type jsonHandler struct {
 }
 
 func (h jsonHandler) Handle(ctx context.Context, treq *transport.Request, rw transport.ResponseWriter) error {
+	treq.Encoding = Encoding
+	// TODO(abg): Should we fail requests if Rpc-Encoding does not match?
```

Comment:
The only reason to make it optional would be to allow people to write catch-all procedures... although I suppose they could register the procedure on every encoding they hope to catch. The same can be said for the `(service, procedure)` tuple, a catch-all service would need to register on every `(service, procedure)` pair they hope to catch...